libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
Currently, the user-visible result of the thread ending is that `sweep_device_get_scan` will deadlock. Afaik that function should signal an error in the worker thread. Although, I think it should be possible to cancel a call to `sweep_device_get_scan` (and/or use a timeout)